src/GovUk.Education.ExploreEducationStatistics.Public.Data.Model/TimePeriodMeta.cs

```diff
@@ -1,10 +1,36 @@
+using GovUk.Education.ExploreEducationStatistics.Common.Converters;
 using GovUk.Education.ExploreEducationStatistics.Common.Model;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Metadata.Builders;
 
 namespace GovUk.Education.ExploreEducationStatistics.Public.Data.Model;
 
-public class TimePeriodMeta
+public class TimePeriodMeta : ICreatedUpdatedTimestamps<DateTimeOffset, DateTimeOffset?>
 {
+    public int Id { get; set; }
+
+    public required Guid DataSetVersionId { get; set; }
+
+    public DataSetVersion DataSetVersion { get; set; } = null!;
+
     public required TimeIdentifier Code { get; set; }
 
     public required int Year { get; set; }
+
+    public DateTimeOffset Created { get; set; }
+
+    public DateTimeOffset? Updated { get; set; }
+
+    internal class Config : IEntityTypeConfiguration<TimePeriodMeta>
+    {
+        public void Configure(EntityTypeBuilder<TimePeriodMeta> builder)
+        {
+            builder.Property(o => o.Code)
+                .HasConversion(new EnumToEnumValueConverter<TimeIdentifier>())
+                .HasMaxLength(10);
```

Comment:
Yeah sounds like a good idea!

Only reason I originally suggested using the same length as for `GeographicLevel` is because I thought the length we chose was arbitrary.

But sounds like it has a decent potential of climbing to longer lengths so I'm happy with 10 :)